x-pack/plugins/security_solution/public/resolver/test_utilities/simulator/index.tsx

```diff
@@ -175,14 +175,24 @@ export class Simulator {
   }
 
   /**
-   * Return an Enzyme ReactWrapper for any child elements of a specific processNodeElement
-   *
-   * @param entityID The entity ID of the proocess node to select in
-   * @param selector The selector for the child element of the process node
+   * The button that opens a node's submenu.
    */
-  public processNodeChildElements(entityID: string, selector: string): ReactWrapper {
+  public processNodeSubmenuButton(
```

Comment:
This no longer relies on the DOM structure.

Reply:
Cool, I wonder if it's better to just remove this method entirely to just use the `testSubject` and make the `processNodeSubmenuButton` a separate utility function? Only rationale is keeping the simulator as generic as possible. so you would have something like `expect(simulator().testSubject(processNodeSubmenuButton(entityId))).toBe(blah)`...then again that looks a bit gross 🤷‍♂️

Reply:
i actually considered this, but the `testSubject` method takes just a single test subject string while `processNodeSubmenuButton` involves two data-attribute selectors. 

I also considered packing all of the info we need into the `data-test-subj` attribute.

I wasn't 100% on either of those, but I think this PR is still an improvement.

Reply:
Yea, it can really go either way. We could have testSubject take an options obj that builds the selector based on what's defined a la your `processNodeElementSelector`. So testSubject would now take the following 
```
{
	dataTestSubj: string,
	dataTestResolverNodeId: string,
    ...potentially any other future options
}

```

Reply:
yah that might work. I first want to discuss this topic: is it important that the buttons are descendants of the node's wrapper `<div />`. Having methods like `processNodeSubmenuButton` provides a level of abstraction that will allow us to easily change our mind on that. In this PR it selects an element using two attribute selectors. If we decide that the DOM structure does matter then we could change that to use a descendant selector.

Reply:
This seems to be resolved by @bkimmel's comment below. Thanks